pkg/liqoctl/virtualnode/create.go

```diff
@@ -0,0 +1,214 @@
+  --cluster-name my-cluster-name --kubeconfig-secret-name my-cluster-kubeconfig --namespace my-cluster`
+
+// Create creates a VirtualNode.
+func (o *Options) Create(ctx context.Context, options *rest.CreateOptions) *cobra.Command {
+	outputFormat := args.NewEnum([]string{"json", "yaml"}, "")
+
+	o.createOptions = options
+
+	cmd := &cobra.Command{
+		Use:     "virtualnode",
+		Aliases: []string{"vn"},
+		Short:   "Create a virtual node",
+		Long:    liqoctlCreateVirtualNodeLongHelp,
+		Args:    cobra.ExactArgs(1),
+
+		PreRun: func(cmd *cobra.Command, args []string) {
+			options.OutputFormat = outputFormat.Value
+			options.Name = args[0]
+			o.createOptions = options
+		},
+
+		Run: func(cmd *cobra.Command, args []string) {
+			output.ExitOnErr(o.handleCreate(ctx))
+		},
+	}
+
+	cmd.Flags().VarP(outputFormat, "output", "o",
+		"Output the resulting VirtualNode resource, instead of applying it. Supported formats: json, yaml")
+
+	// TODO: check validity of both cluster-id and cluster-name
+	cmd.Flags().StringVar(&o.remoteClusterIdentity.ClusterID, "cluster-id", "", "The cluster ID of the remote cluster")
+	cmd.Flags().StringVar(&o.remoteClusterIdentity.ClusterName, "cluster-name", "", "The cluster name of the remote cluster")
+	cmd.Flags().BoolVar(&o.createVirtualNode, "create-node",
+		true, "Create a node to target the remote cluster (and schedule on it)")
+	cmd.Flags().StringVar(&o.kubeconfigSecretName, "kubeconfig-secret-name",
+		"", "The name of the secret containing the kubeconfig of the remote cluster")
+	cmd.Flags().StringVar(&o.cpu, "cpu", "2", "The amount of CPU available in the virtual node")
+	cmd.Flags().StringVar(&o.memory, "memory", "4Gi", "The amount of memory available in the virtual node")
+	cmd.Flags().StringVar(&o.pods, "pods", "110", "The amount of pods available in the virtual node")
+	cmd.Flags().StringSliceVar(&o.storageClasses, "storage-classes",
+		[]string{}, "The storage classes offered by the remote cluster. The first one will be used as default")
+	cmd.Flags().StringToStringVar(&o.labels, "labels", map[string]string{}, "The labels to be added to the virtual node")
+
+	runtime.Must(cmd.MarkFlagRequired("cluster-id"))
+	runtime.Must(cmd.MarkFlagRequired("cluster-name"))
+	runtime.Must(cmd.MarkFlagRequired("kubeconfig-secret-name"))
+
+	runtime.Must(cmd.RegisterFlagCompletionFunc("cluster-id", completion.ClusterIDs(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+	runtime.Must(cmd.RegisterFlagCompletionFunc("cluster-name", completion.ClusterNames(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+	runtime.Must(cmd.RegisterFlagCompletionFunc("kubeconfig-secret-name", completion.KubeconfigSecretNames(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+
+	return cmd
+}
+
+func (o *Options) handleCreate(ctx context.Context) error {
+	opts := o.createOptions
+	if opts.OutputFormat != "" {
+		opts.Printer.CheckErr(o.output())
+		return nil
+	}
```

Comment:
Sorry, I don't understand what's the purpose of this piece of code. Why you are printing the resource in the create command?